go-runtime/compile/build.go

```diff
@@ -98,16 +99,39 @@ func Build(ctx context.Context, moduleDir string, sch *schema.Schema) error {
 		return fmt.Errorf("failed to generate external modules: %w", err)
 	}
 
+	buildDir := buildDir(moduleDir)
 	logger.Debugf("Extracting schema")
 	nativeNames, main, err := ExtractModuleSchema(moduleDir)
 	if err != nil {
-		return fmt.Errorf("failed to extract module schema: %w", err)
+		var schemaErrs []*schema.Error
+		var otherErrs []error
+		for _, e := range errors.DeduplicateErrors(errors.UnwrapAll(err)) {
+			var ce schema.Error
+			if errors.As(e, &ce) {
+				schemaErrs = append(schemaErrs, &ce)
+			} else {
+				otherErrs = append(otherErrs, e)
+			}
+		}
+		el := schema.ErrorList{
+			Errors: schemaErrs,
+		}
+		elBytes, err := proto.Marshal(el.ToProto())
+		if err != nil {
+			return fmt.Errorf("failed to marshal errors: %w", err)
+		}
+
+		err = os.WriteFile(filepath.Join(buildDir, "errors.pb"), elBytes, 0600)
+		if err != nil {
+			return fmt.Errorf("failed to write errors: %w", err)
+		}
+
+		return fmt.Errorf("failed to extract module schema: %w", errors.Join(otherErrs...))
```

Comment:
If `otherErrs` is empty this will return a weird error, but also doesn't this conflict with the logic above where we only load the `errors.pb` if this function returns an error? Maybe we want to return the original `err` here instead?